### PR TITLE
`Development`: Speed up client tests for programming exercise grading configuration

### PR DESCRIPTION
**Pull request by b-fein**
#### General
- [x] Test run locally.
- [x] Test run locally when running with leak detection.
- [x] When repeatedly running the whole test suite locally this test no longer takes >500 seconds and instead finishes fairly quickly.

#### Code Review
- [x] Review 1
- [ ] Review 2

### Test Coverage
The chart components are now mocked and it no longer tested that they produce the correct tooltips. Now only the data they are called with is tested.

**The Bamboo client tests currently fail due to the now slightly insufficient coverage or due to another slow test, but no longer time out because this test is slow.**
```
Jest: "global" coverage threshold for branches (63.5%) not met: 63.46%
```
When merged together with #4263 the coverage should no longer be an issue.

@willergeorg and I will move the chart specific testing of `CategoryIssuesChartComponent` to a new test suite and check if it causes those huge slowdowns there as well. If not, then the coverage will improve to at least the one like before.
A separate test suite for the `TestCasePassedBuildsChartComponent` has been introduced already in this PR, but this is not enough to successfully hit the coverage goal. 

# Bamboo Timeouts Still Not Resolved
When pushing a commit for this PR I noticed that Bamboo still timed out. Both test suites relevant to this PR passed. It always crashes when only a single test has not yet passed. This test is a random one and different on every run.
I suspect that there is some problem between shutting down the worker threads after testing and then generating the coverage report that takes so long that Bamboo decides to kill the task.
I cannot replicate this issue by running the tests locally in the same docker image as Bamboo.

### CODE REVIEW

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-configure-grading.component.spec.ts [hunk 1]
```diff
@@ -201,23 +206,23 @@ describe('ProgrammingExerciseConfigureGradingComponent', () => {
         return getElement(debugElement, testCasesNoUpdated);
     };
 
-    beforeEach(async(() => {
+    beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [TranslateModule.forRoot(), ArtemisTestModule, NgxDatatableModule, NgbModule],
+            imports: [TranslateTestingModule, ArtemisTestModule, NgxDatatableModule, NgbModule],
```
Comment (willergeorg) [inline src/test/javascript/spec/component/programming-exercise/programming-exercise-configure-grading.component.spec.ts:211]: If possible, mock the `TranslateTestingModule`. Also have a look whether the `NgbModule` is actually needed. Might be possible to mock some parts of it. I think the `AlertService` can be mocked as well.

**Review by willergeorg — COMMENTED** [review]
Code changes look good in general 👍. I left one small inline comment.

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-configure-grading.component.spec.ts (hunk 1 above)
Comment (b-fein) [reply src/test/javascript/spec/component/programming-exercise/programming-exercise-configure-grading.component.spec.ts]: Thanks, all three of those suggestions were possible actually. I implemented them in 83c02e2dd0eb841bf88d23cbc222584d4e1e94f0.

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-configure-grading.component.spec.ts [hunk 2]
```diff
@@ -201,31 +206,33 @@ describe('ProgrammingExerciseConfigureGradingComponent', () => {
...
         TestBed.configureTestingModule({
-            imports: [TranslateModule.forRoot(), ArtemisTestModule, NgxDatatableModule, NgbModule],
+            imports: [ArtemisTestModule, NgxDatatableModule, MockModule(TranslateTestingModule)],
             declarations: [
                 ProgrammingExerciseConfigureGradingComponent,
                 ProgrammingExerciseConfigureGradingStatusComponent,
-                ProgrammingExerciseConfigureGradingActionsComponent,
                 ProgrammingExerciseGradingTableActionsComponent,
+                TableEditableFieldComponent,
+                NgbPopover,
```
Comment (willergeorg) [inline src/test/javascript/spec/component/programming-exercise/programming-exercise-configure-grading.component.spec.ts:217]:
```suggestion
                MockDirective(NgbPopover),
```

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-configure-grading.component.spec.ts [hunk 3]
```diff
@@ -201,31 +206,33 @@ describe('ProgrammingExerciseConfigureGradingComponent', () => {
...
+                NgbTooltip,
```
Comment (willergeorg) [inline src/test/javascript/spec/component/programming-exercise/programming-exercise-configure-grading.component.spec.ts:218]:
```suggestion
                MockDirective(NgbTooltip),
```

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-test-case-passed-builds-charts.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import { ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';
+import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
```
Comment (willergeorg) [inline src/test/javascript/spec/component/programming-exercise/programming-exercise-test-case-passed-builds-charts.component.spec.ts:2]:
```suggestion
import { MockDirective } from 'ng-mocks';
import { NgbTooltip } from '@ng-bootstrap/ng-bootstrap';
```

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-test-case-passed-builds-charts.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import { TestCaseStats } from 'app/entities/programming-exercise-test-case-statistics.model';
+
+describe('TestCasePassedBuildsChartComponent', () => {
+    let comp: TestCasePassedBuildsChartComponent;
+    let fixture: ComponentFixture<TestCasePassedBuildsChartComponent>;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [NgbModule],
+            declarations: [TestCasePassedBuildsChartComponent],
```
Comment (willergeorg) [inline src/test/javascript/spec/component/programming-exercise/programming-exercise-test-case-passed-builds-charts.component.spec.ts:14]:
```suggestion
            imports: [],
            declarations: [TestCasePassedBuildsChartComponent, MockDirective(NgbTooltip)],
```

**Review by willergeorg — APPROVED** [review]
Only a few small nits remained. Besides that, lgtm.

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-configure-grading.component.spec.ts (hunk 2 above)
Comment (b-fein) [reply src/test/javascript/spec/component/programming-exercise/programming-exercise-configure-grading.component.spec.ts]: Thanks, now I understand why mocking with `MockComponent` didn’t work. :smile:

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-test-case-passed-builds-charts.component.spec.ts (hunk 5 above)
Comment (b-fein) [reply src/test/javascript/spec/component/programming-exercise/programming-exercise-test-case-passed-builds-charts.component.spec.ts]: Applied together with the other suggestions. :+1:

**Review by MDK5 — APPROVED** [review]
Reviewed code, changes look good to me

**Pull request merged**